cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java

```diff
@@ -32,21 +32,30 @@
 import java.util.Map;
 import java.util.Objects;
 import java.util.Properties;
+import mil.army.usace.hec.cwms.http.client.request.HttpPostRequest;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestExecutor;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestMediaType;
+import mil.army.usace.hec.cwms.http.client.request.HttpRequestMethod;
 import okhttp3.HttpUrl;
+import okhttp3.MediaType;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
+import okhttp3.RequestBody;
 import okhttp3.Response;
 import okhttp3.ResponseBody;
 import usace.metrics.noop.NoOpTimer;
 import usace.metrics.services.Metrics;
 import usace.metrics.services.Timer;
 
-public class HttpRequestBuilderImpl implements HttpRequestBuilder {
+public final class HttpRequestBuilderImpl implements HttpRequestBuilder {
 
     private final String endpoint;
     private final HttpUrl httpUrl;
     private final Map<String, String> queryParameters = new HashMap<>();
     private final Map<String, String> queryHeaders = new HashMap<>();
+    private HttpRequestMethod method;
+    private String body;
+    private String mediaType;
```

Comment:
How varied is this media type? Can this be an enum?

Reply:
Good question. Looks like there's quite alot of formats that are supported in MediaType.parse(String). We are only supporting different json versions (and possibly some xml), currently however.

Reply:
Ok, String is file with me